perception/lidar_centerpoint_tvm/lib/postprocess/generate_detected_boxes.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 AutoCore Ltd., TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lidar_centerpoint_tvm/postprocess/generate_detected_boxes.hpp"
+
+#include <lidar_centerpoint_tvm/postprocess/circle_nms.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <thread>
+
+namespace
+{
+const std::size_t THREAD_NUM_POST = 32;
+}  // namespace
+
+namespace autoware
+{
+namespace perception
+{
+namespace lidar_centerpoint_tvm
+{
+
+struct is_score_greater
+{
+  is_score_greater(float32_t t) : t_(t) {}
+  bool8_t operator()(const Box3D & b) { return b.score > t_; }
+
+private:
+  float32_t t_{0.0};
+};
+
+struct is_kept
+{
+  bool8_t operator()(const bool8_t keep) { return keep; }
+};
+
+struct score_greater
+{
+  bool operator()(const Box3D & lb, const Box3D & rb) { return lb.score > rb.score; }
+};
+
+inline float32_t sigmoid(float32_t x) { return 1.0f / (1.0f + expf(-x)); }
+
+void generateBoxes3D_worker(
+  const float32_t * out_heatmap, const float32_t * out_offset, const float32_t * out_z,
+  const float32_t * out_dim, const float32_t * out_rot, const float32_t * out_vel,
+  const CenterPointConfig & config, Box3D * boxes3d, std::size_t thread_idx,
+  std::size_t grids_per_thread)
+{
+  // generate boxes3d from the outputs of the network.
+  // shape of out_*: (N, DOWN_GRID_SIZE_Y, DOWN_GRID_SIZE_X)
+  // heatmap: N = class_size, offset: N = 2, z: N = 1, dim: N = 3, rot: N = 2, vel: N = 2
+  for (std::size_t idx = 0; idx < grids_per_thread; idx++) {
+    std::size_t grid_idx = thread_idx * grids_per_thread + idx;
+    const auto down_grid_size = config.down_grid_size_y_ * config.down_grid_size_x_;
+    if (grid_idx >= down_grid_size) return;
+
+    const auto yi = (grid_idx % down_grid_size) / config.down_grid_size_x_;
+    const auto xi = (grid_idx % down_grid_size) / config.down_grid_size_y_;
```

Comment:
By doing this, `xi` and `yi` are always equal because `down_grid_size = down_grid_size_y_ = down_grid_size_x_`.
One way to imitate `xi` and `yi` as it used to do is like the following:
```
    const auto xi = grid_idx % config.down_grid_size_x_;
    const auto yi = grid_idx / config.down_grid_size_x_;
```

Reply:
yes, it has been modified.